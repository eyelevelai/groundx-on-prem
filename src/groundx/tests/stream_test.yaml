suite: "stream renders as expected"
values:
  - ../values.yaml

tests:
  - it: stream_cluster.yaml
    templates:
      - ../templates/services/stream_cluster.yaml
    asserts:
      - matchSnapshot: {}
  - it: stream_topics.yaml
    templates:
      - ../templates/services/stream_topics.yaml
    asserts:
      - matchSnapshot: {}