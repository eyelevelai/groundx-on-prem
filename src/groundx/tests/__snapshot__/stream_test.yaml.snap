stream_cluster.yaml:
  1: |
    apiVersion: kafka.strimzi.io/v1beta2
    kind: KafkaNodePool
    metadata:
      labels:
        app: stream-dual-role
        appVersion: 0.1.0
        chart: groundx-0.1.0
        heritage: Helm
        strimzi.io/cluster: stream-cluster
        version: 0.1.0
      name: stream-dual-role
      namespace: eyelevel
    spec:
      replicas: 1
      roles:
        - controller
        - broker
      storage:
        type: jbod
        volumes:
          - deleteClaim: true
            id: 0
            kraftMetadata: shared
            size: 5Gi
            type: persistent-claim
  2: |
    apiVersion: kafka.strimzi.io/v1beta2
    kind: Kafka
    metadata:
      annotations:
        strimzi.io/kraft: enabled
        strimzi.io/node-pools: enabled
      labels:
        app: stream
        appVersion: 0.1.0
        chart: groundx-0.1.0
        heritage: Helm
        version: 0.1.0
      name: stream-cluster
      namespace: eyelevel
    spec:
      entityOperator:
        topicOperator: {}
        userOperator: {}
      kafka:
        config:
          default.replication.factor: 1
          inter.broker.protocol.version: 4.0-IV3
          log.message.format.version: 4.0-IV3
          min.insync.replicas: 1
          offsets.topic.replication.factor: 1
          transaction.state.log.min.isr: 1
          transaction.state.log.replication.factor: 1
          unstable.api.versions.enable: "true"
          unstable.feature.versions.enable: "true"
        listeners:
          - name: kafka
            port: 9092
            tls: false
            type: internal
        metadataVersion: 4.0-IV3
        replicas: 1
        resources:
          requests:
            cpu: 200m
            memory: 512Mi
        storage:
          deleteClaim: true
          size: 10Gi
          type: persistent-claim
        template:
          pod:
            affinity:
              nodeAffinity:
                requiredDuringSchedulingIgnoredDuringExecution:
                  nodeSelectorTerms:
                    - matchExpressions:
                        - key: node
                          operator: In
                          values:
                            - eyelevel-cpu-only
            tolerations:
              - effect: NoSchedule
                key: node
                value: eyelevel-cpu-only
        version: 4.0.0
stream_topics.yaml:
  1: |
    apiVersion: kafka.strimzi.io/v1beta2
    kind: KafkaTopic
    metadata:
      labels:
        strimzi.io/cluster: stream-cluster
      name: file-pre-process
    spec:
      config:
        retention.bytes: 1073741824
        retention.ms: 7200000
        segment.bytes: 1073741824
      partitions: 1
      replicas: 1
  2: |
    apiVersion: kafka.strimzi.io/v1beta2
    kind: KafkaTopic
    metadata:
      labels:
        strimzi.io/cluster: stream-cluster
      name: file-process
    spec:
      config:
        retention.bytes: 1073741824
        retention.ms: 7200000
        segment.bytes: 1073741824
      partitions: 1
      replicas: 1
  3: |
    apiVersion: kafka.strimzi.io/v1beta2
    kind: KafkaTopic
    metadata:
      labels:
        strimzi.io/cluster: stream-cluster
      name: file-summary
    spec:
      config:
        retention.bytes: 1073741824
        retention.ms: 7200000
        segment.bytes: 1073741824
      partitions: 1
      replicas: 1
  4: |
    apiVersion: kafka.strimzi.io/v1beta2
    kind: KafkaTopic
    metadata:
      labels:
        strimzi.io/cluster: stream-cluster
      name: file-update
    spec:
      config:
        retention.bytes: 1073741824
        retention.ms: 7200000
        segment.bytes: 1073741824
      partitions: 1
      replicas: 1
  5: |
    apiVersion: kafka.strimzi.io/v1beta2
    kind: KafkaTopic
    metadata:
      labels:
        strimzi.io/cluster: stream-cluster
      name: file-upload
    spec:
      config:
        retention.bytes: 1073741824
        retention.ms: 7200000
        segment.bytes: 1073741824
      partitions: 1
      replicas: 1
